File: 06_strings/06_04_longest_common_prefix/longest_common_prefix.py
def longest_common_prefix(args: list[str]):
    if not len(args):
        return ''
    
    args.sort()
    index = 0
    for i in range(len(args[0])):
        if args[0][i] == args[-1][i]:
            index += 1
        else:
            break
    return args[0][:index]

# Test case 1: Basic example
input1 = ["flower", "flow", "flight"]
result1 = longest_common_prefix(input1)
print(f"Test Case 1: Input: {input1}, Result: \"{result1}\"")

# Test case 2: No common prefix
input2 = ["dog", "racecar", "car"]
result2 = longest_common_prefix(input2)
print(f"Test Case 2: Input: {input2}, Result: \"{result2}\"")

# Test case 3: Common prefix at the beginning
input3 = ["apple", "apricot", "apology"]
result3 = longest_common_prefix(input3)
print(f"Test Case 3: Input: {input3}, Result: \"{result3}\"")

# Test case 4: Empty array
input4 = []
result4 = longest_common_prefix(input4)
print(f"Test Case 4: Input: {input4}, Result: \"{result4}\"")

# Test case 5: Single-word array
input5 = ["programming"]
result5 = longest_common_prefix(input5)
print(f"Test Case 5: Input: {input5}, Result: \"{result5}\"")

File: 06_strings/06_04_longest_common_prefix/LongestCommonPrefix.java
import java.util.Arrays;

public class LongestCommonPrefix {
    public static void main(String[] args) {
        // Test case 1: Basic example
        String[] input1 = { "flower", "flow", "flight" };
        String result1 = longestCommonPrefix(input1);
        System.out.println("Test Case 1: Input: " + Arrays.toString(input1) + ", Result: \"" + result1 + "\"");

        // Test case 2: No common prefix
        String[] input2 = { "dog", "racecar", "car" };
        String result2 = longestCommonPrefix(input2);
        System.out.println("Test Case 2: Input: " + Arrays.toString(input2) + ", Result: \"" + result2 + "\"");

        // Test case 3: Common prefix at the beginning
        String[] input3 = { "apple", "apricot", "apology" };
        String result3 = longestCommonPrefix(input3);
        System.out.println("Test Case 3: Input: " + Arrays.toString(input3) + ", Result: \"" + result3 + "\"");

        // Test case 4: Empty array
        String[] input4 = {};
        String result4 = longestCommonPrefix(input4);
        System.out.println("Test Case 4: Input: " + Arrays.toString(input4) + ", Result: \"" + result4 + "\"");

        // Test case 5: Single-word array
        String[] input5 = { "programming" };
        String result5 = longestCommonPrefix(input5);
        System.out.println("Test Case 5: Input: " + Arrays.toString(input5) + ", Result: \"" + result5 + "\"");
    }

    private static String longestCommonPrefix(String[] args) {
        if (args.length <= 0) {
            return "";
        }
        Arrays.sort(args);

        int length = args[0].length();
        int i = 0;
        while (i < length) {
            if (args[0].charAt(i) == args[args.length - 1].charAt(i)) {
                i++;
            } else {
                break;
            }
        }

        return args[0].substring(0, i);
    }
}
